test/test_basic_logic.py

```diff
@@ -439,8 +439,16 @@ def test_dict_headers(self):
         """
         Sending headers using dict is deprecated but still valid.
         """
+        # One of the steps in the outbound header validation logic checks
+        # that we don't send a pseudo-header after we've sent a regular
+        # header.  We can't guarantee that if you send headers as a dict,
+        # because dicts are unordered.
+        config = h2.config.H2Configuration(
+            validate_outbound_headers=False
+        )
+
```

Comment:
Yes, I thought carefully about this, and I’m in two minds about whether to actually put this in now, or wait for 3.0.

Other checks in this block assert that you include :authority or Host headers, and if you choose the former then you’re stuffed: depending on your Python, this could randomly explode on you. Which feels less like deprecation and more like deletion.

I’ve put in a changelog entry for it, but I’m tempted to remove it now and revisit it for 3.0.

What do you think?